### PR TITLE
test: fix flakiness in CI

### PR DESCRIPTION
**Pull request by aboedo**
`BackendIntegrationTests` are flaky right now in CI, but they're fine locally. We get failures from different tests on each run. 

I think this may be happening because `BackendIntegrationTests` started using SKTest, and SKTest needs time to initialize, but we didn't add that to `BackendIntegrationTests`. 

We do have an artificial delay for `StoreKitConfigTestCase`. 

So this ugly PR just makes `BaseBackendIntegrationTests` a subclass of `StoreKitConfigTestCase` so that it gets this artificial delay.

### CODE REVIEW

**Review by NachoSoto — COMMENTED** [review]
This is helpful though 👍🏻 

One thing we need to disable in integration tests is the empty `Transaction` listener in `StoreKitConfigTestCase.setUp` though, I think that'll break the "production" code?
Also I think we need to add `CIRCLECI_STOREKIT_TESTS_DELAY_SECONDS` to the Integration tests, or are the environment variables shared across tests?

**Comment by NachoSoto** [comment]
I've also pushed #1487 and #1488 to help debug these. This run failed with:
> ✖ testPurchaseMadeBeforeLogInWithExistingUserIsNotRetainedUnlessRestoreCalled, failed - expected to equal <1>, got <0>

Which is hard to debug.

**Comment by aboedo** [comment]
> Also I think we need to add CIRCLECI_STOREKIT_TESTS_DELAY_SECONDS to the Integration tests, or are the environment variables shared across tests?

Env variables are repo-wide, so we should be good there

>One thing we need to disable in integration tests is the empty Transaction listener in StoreKitConfigTestCase.setUp though, I think that'll break the "production" code?

You should be able to have multiple observers, right? If not then we should dig deeper into that, since we'd have to warn customers to ensure that there's no observer set up anywhere else other than the SDK if using SK2

**Comment by NachoSoto** [comment]
> You should be able to have multiple observers, right? If not then we should dig deeper into that, since we'd have to warn customers to ensure that there's no observer set up anywhere else other than the SDK if using SK2

Oh yeah I think you're right

File: RevenueCat.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -119,6 +119,8 @@
 		2DE20B8926409EB8004C597D /* Preview Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 2DE20B8826409EB8004C597D /* Preview Assets.xcassets */; };
 		2DE20B9226409ECF004C597D /* StoreKit.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 2DE20B9126409ECF004C597D /* StoreKit.framework */; };
 		2DE21DE928046EAA002297A4 /* StoreKitConfigTestCase.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2D90F8CB26FD2BA1009B9142 /* StoreKitConfigTestCase.swift */; };
+		2DE21DEC280470B0002297A4 /* UnitTestsConfiguration.storekit in Resources */ = {isa = PBXBuildFile; fileRef = 2D43017726EBFD7100BAB891 /* UnitTestsConfiguration.storekit */; };
```
Comment (NachoSoto) [inline RevenueCat.xcodeproj/project.pbxproj:122]: This is now in integration tests? I'm so confused, Integration tests use `RevenueCat_IntegrationPurchaseTesterConfiguration.storekit`.

**Comment by NachoSoto** [comment]
FYI integration tests are no longer flaky in `main` 🎉

**Comment by aboedo** [comment]
@NachoSoto I'm gonna close this out in that case, this was just a test to see if introducing the artificial delay would fix test flakiness in CI
